packages/esm-patient-list-app/src/offline/offlineData.ts

```diff
@@ -0,0 +1,239 @@
+    } else {
+      // creating patient entry
+      updatePromise = this.offlinePatients.add({
+        uuid: patientUuid,
+        name,
+        lastUpdate: Object.fromEntries(getOfflineHandlers().map((handlerUuid) => [handlerUuid, { type: 'LOADING' }])),
+        interestedUsers: [this.currentUser],
+      });
+    }
+    updatePromise.then(() => this.handleDBChange());
+  }
+
+  /**
+   * Remove a patient from the offline list
+   * Triggers remove on all registered handlers if there are no interested users left
+   */
+  async removePatient(patientUuid: string) {
+    const patient = await this.offlinePatients.where({ uuid: patientUuid }).first();
+    if (patient && patient.interestedUsers.includes(this.currentUser)) {
+      if (patient.interestedUsers.length > 1) {
+        return this.offlinePatients.put(
+          { ...patient, interestedUsers: patient.interestedUsers.filter((u) => u !== this.currentUser) },
+          patient.uuid,
+        );
+      } else {
+        Object.values(patientStore.getState()).forEach((handlers) => {
+          handlers.onRemovePatient(patientUuid);
+        });
+        const updatePromise = this.offlinePatients.delete(patient.uuid);
+        updatePromise.then(() => this.handleDBChange());
+        return updatePromise;
+      }
+    }
+  }
+
+  /**
+   * For handlers
+   * Set a resource for a given patient to loaded
+   */
+  async setLastUpdated(patientUuid: string, handlerUuid: string) {
+    const pat = await db.offlinePatients.where({ uuid: patientUuid }).first();
+    if (pat) {
+      const updatePromise = db.offlinePatients.put(
+        {
+          ...pat,
+          lastUpdate: { ...pat.lastUpdate, [handlerUuid]: { type: 'LOADED', date: new Date() } },
+        },
+        pat.uuid,
+      );
+      updatePromise.then(() => this.handleDBChange());
+      return updatePromise;
+    }
+  }
+
+  /**
+   * For handlers
+   * An error occurred while loading.
+   * The last loaded date will be preserved under the assumption that the cache wasn't corrupted
+   */
+  async setUpdateFailed(patientUuid: string, handlerUuid: string, error: Error) {
+    const pat = await db.offlinePatients.where({ uuid: patientUuid }).first();
+    if (pat) {
+      let newLoadState: LoadState;
+      const prevState = pat.lastUpdate[handlerUuid];
+      switch (prevState.type) {
+        case 'LOADING':
+          newLoadState = { type: 'LOADING_ERROR', error };
+          break;
+
+        case 'RELOADING':
+          newLoadState = { type: 'RELOADING_ERROR', error, date: prevState.date };
+          break;
+
+        default:
+          return;
+      }
+      const updatePromise = db.offlinePatients.put(
+        { ...pat, lastUpdate: { ...pat.lastUpdate, [handlerUuid]: newLoadState } },
+        pat.uuid,
+      );
+      updatePromise.then(() => this.handleDBChange());
+      return updatePromise;
+    }
+  }
+}
+
+const db = new OfflinePatientDatabase();
+
+export default {
+  addPatient: db.addPatient,
```

Comment:
Do these work? Because here we are not capturing the `db` object and thus we might have a problem.